src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GeneratorScalingEntity.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.modification.server.entities.equipment.modification;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import org.gridsuite.modification.server.ModificationType;
+import org.gridsuite.modification.server.dto.GeneratorScalingInfos;
+import org.gridsuite.modification.server.dto.ModificationInfos;
+import org.gridsuite.modification.server.dto.ScalingVariationInfos;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity
+@EqualsAndHashCode
+@Table(name = "GeneratorScaling")
+public class GeneratorScalingEntity extends ScalingEntity {
+
+    @Column(name = "isIterative")
+    private boolean isIterative;
+
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
+    private List<ScalingVariationEntity> variations;
+
+    public GeneratorScalingEntity(@NotNull GeneratorScalingInfos generatorScalingInfos) {
+        super(ModificationType.GENERATOR_SCALING);
+        assignAttributes(generatorScalingInfos);
+    }
+
+    @Override
+    public void update(@NonNull ModificationInfos modificationInfos) {
+        super.update(modificationInfos);
+        assignAttributes((GeneratorScalingInfos) modificationInfos);
+    }
+
+    private void assignAttributes(GeneratorScalingInfos generatorScalingInfos) {
```

Comment:
Add methods  `update` and `assignAttributes`  in the superclass  `ScalingEntity` for attribute `variationType`

Reply:
done